.github/workflows/generate-docs.yml

```diff
@@ -0,0 +1,94 @@
+name: Generate docs
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        type: string
+        required: true
+        description: "Documentation version to generate"
+      source-repo:
+        type: string
+        required: true
+        description: "Repository to build from"
+      source-ref:
+        type: string
+        required: true
+        description: "Git ref of the source repo"
+      output-prefix:
+        type: string
+        required: true
+        description: "Output path prefix"
+      auto-merge:
+        type: boolean
+        required: false
+        default: false
+        description: "Whether to automatically merge the branch with the generated docs. If false, creates a pull request instead."
+
+jobs:
+
+  generate-docs:
+    name: Generate docs
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Checkout ${{ github.event.inputs.source-repo }}
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ github.event.inputs.source-repo }}
+          ref: ${{ github.event.inputs.source-ref }}
+          path: source
+          fetch-depth: 0
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+          architecture: x64
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install -r source/requirements.txt
+      - name: Configure Git
+        run: |
+          git config user.name "afakebot"
+          git config user.email "afakebot@users.noreply.github.com"
+      - name: Get latest tag
+        id: get-latest-tag
+        uses: ./.github/actions/latest-stable-tag
+        with:
+          repo-path: source
+      - name: Get source commit
+        id: get-source-commit
+        working-directory: source
+        run: |
+          echo "sha1=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+      - name: Generate the docs
+        id: generate-docs
+        run: |
+          version=${{ github.event.inputs.version }}
+          stable=
+          message="Generate docs from ${{ github.event.inputs.source-repo }}@${{ steps.get-source-commit.outputs.sha1 }}"
+          if [ "$version" = "${{ steps.get-latest-tag.outputs.latest-tag }}" ]; then
+            stable="stable"
+            message="$message (stable)"
+          fi
+          targetBranch=update-$version-docs
+          echo "docs-branch=$targetBranch" >> $GITHUB_OUTPUT
+          git branch $targetBranch
+          mike deploy "$version" $stable \
```

Comment:
Oh! I never realized that option was needed. I will add it.
I tried to regenerate 7.3.1, and it seemed to run the right command (with the "stable" alias). I couldn't see any diff in stable/, but I assumed it was because it was still redirecting to the same version as before... And I couldn't try with e.g. 7.3.2 without creating a tag on FakeItEasy, which would have triggered a release.

Reply:
It should be fixed now

Reply:
I think so too. Yeah, the problem crops up when shifting stable from 7.3.1 to 7.3.2, for example.